### DISCUSSION TITLE
[IMPORTANT] Working dev branch!

### DISCUSSION THREAD

**Discussion by jacueblol** [Patribots4738/Crescendo2024#135] [Announcements]
@Patribots4738/students,

Through some discussion, we have decided to base everything currently in development on a new branch called `sim-dev.` The basis behind this change is to avoid what happened with #129 and #100 merging into the main before completion. Overall your workflow will not change except for the final final step, which is more important for the reviewer than the assignee in most cases: 

## IMPORTANT 

Make a PR to `sim-dev` (not `main`). Then, once you are **positive** that everything is working in sim, you can make another PR from `sim-dev` to `main`. After that is approved, **please remember to restore the branch after the merge**. If it auto removes the branch, just click the `restore` button at the bottom of the PR.

Your workflow should be as such:
- Make a branch for a new feature or bug request 
- Once you are complete and you think that everything works, test it in the simulation. 
- After testing it and proving that it works, make a pr to merge it into `sim-dev`
- One last time, integrate and finalize everything about your new feature or bug in sim, hence the name of the branch. This is our final step before preparing to test on a real robot.
- Once you are sure that everything is working correctly, merge from `sim-dev` to `main`, however, do not remove `sim-dev`

**Comment by tina-hague** [comment]
I remember asking about this yesterday. I think 100 & 129 were merged into main _before_ it was tested but hadizah and alex probably remember better than me. Is the dev branch only meant to prevent this from happening in main or is there other reasons? Do bugs show up in main frequently or is that only when there was an untested branch that got merged?

**Reply by jacueblol** [reply]
The point of this branch is to stop any errors that haven't been tested to stop from going into main first

**Reply by jacueblol** [reply]
We rarely get bugs in main but there are occasional merges that happen

**Reply by GalexY727** [reply]
#100 was not tested before going into main, correct. Yet it got into main  : D

**Reply by jacueblol** [reply]
correct which is y i suggest we make rules and make dev the default #153

**Comment by tina-hague** [comment]
This also seems like a good way to use main as precedence when messing with the dev branch  so we already know what works

**Comment by jacueblol** [comment]
The final thing that we landed on for this was a great way to get the code started but it got cumbersome after a while.
